### ISSUE TITLE
Allow `cancel deployment` to make a FAILING deployment CANCELED

### ISSUE THREAD

**Issue by reid47** [cloudfoundry/cloud_controller_ng#1364]
Currently if we try to cancel a FAILING deployment we get an error message:

 ```
cf curl -X POST /v3/deployments/DEPLOYMENT_GUID/actions/cancel
``` 
we get this error message: `"Cannot cancel a FAILING deployment"`

It should make the deployment as `CANCELED`

**Comment by reid47** [comment]
After looking at the code, it seems like cancelling a `FAILING` deployment should mark it as `CANCELING`, which will then later get marked as `CANCELED`.

**Comment by tcdowney** [comment]
This was implemented in 10205a09ac3b871f3b066654c9c78a8ef9020a75 and shipped, but the need for it was obviated by the [changes we made to Deployment#state](https://lists.cloudfoundry.org/g/cf-dev/message/8648?p=,,,20,0,0,0::Created,,deployment+state,20,2,0,32131351). These shipping in capi-release 1.84.0.
